tests/test-class-amp-theme-support.php

```diff
@@ -212,6 +212,54 @@ public function test_prepare_response() {
 		$this->assertInstanceOf( 'DOMAttr', $removed_nodes['onclick'] );
 	}
 
+	/**
+	 * Test validate_non_amp_theme.
+	 *
+	 * @global WP_Widget_Factory $wp_widget_factory
+	 * @global WP_Scripts $wp_scripts
+	 * @covers AMP_Theme_Support::validate_non_amp_theme()
+	 */
+	public function test_validate_non_amp_theme() {
+		global $wp_widget_factory, $wp_scripts;
+		$wp_scripts = null;
+
+		add_theme_support( 'amp' );
+		AMP_Theme_Support::init();
+		AMP_Theme_Support::finish_init();
+		$wp_widget_factory = new WP_Widget_Factory();
+		wp_widgets_init();
```

Comment:
Since no widgets are being tested in this response, I think the references to `$wp_widget_factory` and `wp_widgets_init()` in this PR can be removed.